test/830_cni_plugin_test.sh

```diff
@@ -48,12 +48,22 @@ assert_raises "exec_on $HOST1 c2 $PING $C1IP"
 # Now remove and start a new container to see if IP address re-use breaks things
 docker_on $HOST1 rm -f c2
 
-docker_on $HOST1 run --net=none --name=c3 -dt $SMALL_IMAGE /bin/sh
+C3=$(docker_on $HOST1 run --net=none --name=c3 -dt $SMALL_IMAGE /bin/sh)
 
 cni_connect $HOST1 c3 <<EOF
 { "name": "weave", "type": "weave-net" }
 EOF
 
+C3IP=$(container_ip $HOST1 c3)
+
 assert_raises "exec_on $HOST1 c1 $PING $C2IP"
 
+
+# Ensure existing containers can reclaim their IP addresses after CNI has been used -- see #2548
+stop_weave_on $HOST1
+weave_on $HOST1 launch-router
+assert "container_ip $HOST1 $C1" "$C1IP"
```

Comment:
This isn't testing the right thing; `container_ip` essentially looks at the veth to see what IP address is attached, while the thing we are interested in is what IPAM has stored internally.

Also the original complaint was spurious messages, which are not examined here.